R/utilities.R

```diff
@@ -4117,3 +4130,700 @@ supplement_maf <- function(incoming_maf,
+                "A total of ",
+                length(
+                   ashm_missing
+                ),
+                " aSHM features are missing:"
+            )
+        )
+        message(
+            paste(
+                ashm_missing,
+                collapse=", "
+            )
+        )
+        message(
+            "They will be set to 0, which may affect model performance."
+        )
+        ashm_matrix[,ashm_missing] <- 0
+    }
+
+    # Combine together the SSM, hotspots, and aSHM
+    assembled_matrix <- bind_cols(
+            ssm_matrix,
+            ashm_matrix
+        ) %>%
+        select(
+            rownames(
+                model$importance
+            )
+        )
+
+    # Make prediction
+    prediction <- predict(
+        model,
+        assembled_matrix,
+        type="Vote"
+    )
+
+    prediction <- bind_cols(
+        prediction,
+        predict(
+                model,
+                assembled_matrix
+            ) %>%
+            as.data.frame %>%
+            `names<-`("is_cFL")
+        ) %>%
+    rownames_to_column("sample_id")
+
+    if(output=="predictions"){
+        return(prediction)
+    }else if (output=="matrix") {
+       return(assembled_matrix)
+    }else if(output=="both"){
+        return(
+            list(
+                predictions = prediction,
+                matrix = assembled_matrix)
+        )
+    }else{
+        stop("Invalid output type. Please specify predictions, matrix, or both.")
+    }
+
+
+}
+
+#' Classify DLBCLs according to genetic subgroups of Chapuy et al.
+#'
+#' Use the feature weights from NMF model to assemble the binary matrix and classify DLBCL tumors based on C0-C5 system of Chapuy et al
+#'
+#' @param these_samples_metadata The metadata data frame that contains sample_id column with ids for the samples to be classified.
+#' @param maf_data The MAF data frame to be used for matrix assembling. At least must contain the first 45 columns of standard MAF format.
+#' @param seg_data The SEG data frame to be used for matrix assembling. Must be of standard SEG formatting, for example, as returned by get_sample_cn_segments.
+#' @param sv_data The SV data frame to be used for matrix assembling. Must be of standard BEDPE formatting, for example, as returned by get_combined_sv.
+#' @param seq_type The seq_type of the samples. Only used to retrerive data through GAMBLR when it is not provided. Defaults to grch37.
+#' @param projection The projection of the samples. Only used to retrerive data through GAMBLR when it is not provided. Defaults to genome.
+#' @param output The output to be returned after prediction is done. Can be one of predictoins, matrix, or both. Defaults to predictions.
+#' @param adjust_ploidy Whether to perform ploidy adjustment for the CNV data. Defaults to TRUE (recommended).
+#'
+#' @return data frame with classification, binary matrix used in classification, or both
+#' @export
+#' @import data.table circlize tidyverse
```

Comment:
Tidyverse is not really importable, see [this](https://github.com/morinlab/GAMBLR/issues/102) for more info. I think this is fine for now, but at some point, I will go over all GAMBLR functions and update this. So maybe something to keep in mind for any future new functions.

Reply:
Thanks!